lib/ansible/inventory/data.py

```diff
@@ -231,7 +231,7 @@ def add_host(self, host, group=None, port=None):
         else:
             h = self.hosts[host]
 
-        if g and host not in g.get_hosts():
+        if g and h not in g.get_hosts():
```

Comment:
`host` here is a string, and `g.get_hosts()` returns host objects.  So the `not in` always failed.